vault/tests/test_vault.py

```diff
@@ -257,6 +257,39 @@ def mock_requests_get(url, *args, **kwargs):
 
         aggregator.assert_service_check(Vault.SERVICE_CHECK_INITIALIZED, status=Vault.CRITICAL, count=1)
 
+    def test_replication_dr_mode(self, aggregator):
+        instance = INSTANCES['main']
+        c = Vault(Vault.CHECK_NAME, {}, [instance])
+
+        # Keep a reference for use during mock
+        requests_get = requests.get
+
+        def mock_requests_get(url, *args, **kwargs):
+            if url == instance['api_url'] + '/sys/health':
+                return MockResponse(
+                    {
+                        'cluster_id': '9e25ccdb-09ea-8bd8-0521-34cf3ef7a4cc',
+                        'cluster_name': 'vault-cluster-f5f44063',
+                        'initialized': False,
+                        'replication_dr_mode': 'secondary',
+                        'replication_performance_mode': 'primary',
+                        'sealed': False,
+                        'server_time_utc': 1529357080,
+                        'standby': True,
+                        'performance_standby': False,
+                        'version': '0.10.2',
+                    },
+                    status_code=200,
+                )
+            return requests_get(url, *args, **kwargs)
+
+        with mock.patch('requests.get', side_effect=mock_requests_get, autospec=True):
+            run_check(c)
+            c.log.debug("Detected vault in replication DR mode, skipping Prometheus metric collection.")
```

Comment:
How come you debug this message again in the test?

Reply:
oops! meant to call it with `assert_called_with` to check the message was logged and we hit that part of the code